coreax/util.py

```diff
@@ -70,23 +68,8 @@ def apply_negative_precision_threshold(
     :param precision_threshold: Positive threshold we compare against for precision
     :return: ``x``, rounded to 0.0 if it is between ``-precision_threshold`` and 0.0
     """
-    # Validate inputs
-    x = coreax.validation.cast_as_type(x=x, object_name="x", type_caster=float)
-    precision_threshold = coreax.validation.cast_as_type(
-        x=precision_threshold, object_name="precision_threshold", type_caster=float
-    )
-    coreax.validation.validate_in_range(
-        x=precision_threshold,
-        object_name="precision_threshold",
-        strict_inequalities=False,
-        lower_bound=0,
-    )
-
     if precision_threshold < 0.0:
-        raise ValueError(
-            f"precision_threshold must be positive; value {precision_threshold} given."
-        )
-
+        raise ValueError("precision_threshold must not be negative.")
     if -precision_threshold < x < 0.0:
```

Comment:
The type hints don't align with the feasible logic here. For example:
```python
import jax.numpy as jnp
import coreax.util

x = jnp.array([-0.1, 1, 2])  # This is an ArrayLike
precision_threshold = 0.2

x_rounded = coreax.util.apply_negative_precision_threshold(x, precision_threshold)
# ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()
```
The output type hint is also inconsistent. Ignoring the conditional statement, `x` will be returned identically to how it was passed, so the output is an `ArrayLike` (which is not necessarily a float).

Reply:
Thanks for catching this - the `ArrayLike` annotation on `x` was wrong - when the function was written, the docstring also states x as a scalar, I've updated the type annotation.